arangod/Graph/AttributeWeightShortestPathFinder.cpp

```diff
@@ -314,16 +346,43 @@ bool AttributeWeightShortestPathFinder::shortestPath(
 }
 
 void AttributeWeightShortestPathFinder::inserter(
-    std::vector<std::unique_ptr<Step>>& result, std::string_view s,
-    std::string_view t, double currentWeight, EdgeDocumentToken&& edge) {
+    std::vector<std::unique_ptr<Step>>& result, std::string_view prevVertexId,
+    std::string_view nextVertexId, double currentWeight,
+    EdgeDocumentToken&& edge) {
   ResourceUsageScope guard(_resourceMonitor, candidateMemoryUsage());
 
-  auto [cand, emplaced] =
-      _candidates.try_emplace(t, arangodb::lazyConstruct([&] {
-                                result.emplace_back(std::make_unique<Step>(
-                                    t, s, currentWeight, std::move(edge)));
-                                return result.size() - 1;
-                              }));
+#ifdef USE_ENTERPRISE
+  if (options().isDisjoint()) {
```

Comment:
This doesn't work either for the same reason as what Michael described above. We should have a separate `_smartValue` for each path, that means here, a local variable in the caller, which is passed by reference to the lambda. It is set the first time it is seen on the path and if it is set and an other value is seen later, return `false`, otherwise `true`.

@mchacki Don't we have this functionality somewhere else already?